### PR TITLE
Added PolHateClassification task

### PR DESCRIPTION
**Pull request by dokato**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

Added PolHateClassification in analogy with ItaHateClassification.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/pol/PolHateClassification.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        rename_dict = dict(zip(keep_cols, ["text", "label"]))
+        remove_cols = [
+            col for col in self.dataset["test"].column_names if col not in keep_cols
+        ]
+        self.dataset = self.dataset.rename_columns(rename_dict)
+        self.dataset = self.dataset.class_encode_column("label")
+        self.dataset = self.dataset.class_encode_column("functionality")
+        self.dataset = self.dataset["test"].train_test_split(
+            test_size=0.5, seed=42, stratify_by_column="functionality"
+        )  # balanced sampling across types of hate speech
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/pol/PolHateClassification.py:72]: 2 things here:

- I think you should stratify on the labels since that's what the classifier will use, and your dataset is imbalanced: `stratify_by_column="label"`. Altough, I find it interesting to stratify on "functionality", I think that if you'd liek to do taht with sklearn, you'll have to concatenate columns "label" and this one and stratify on it.
- You can use: `self.seed` for the seed value.

File: mteb/tasks/Classification/pol/PolHateClassification.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        self.dataset = self.dataset.remove_columns(remove_cols)
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/pol/PolHateClassification.py:73]: It's good to remove unused columns, but I'm not sure it's useful, since the evaluator will only look at `text` and `label` columns.

File: mteb/tasks/Classification/pol/PolHateClassification.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+
+if __name__ == "__main__":
+    from sentence_transformers import SentenceTransformer
+    from mteb import MTEB
+    # Define the sentence-transformers model name
+    model_name = "sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2"
+    model = SentenceTransformer(model_name)
+    evaluation = MTEB(tasks=[PolHateClassification()])
+    evaluation.run(model)
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/pol/PolHateClassification.py:83]: Can you please run models : `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2` and `intfloat/multilingual-e5-small` on your task, and push the results along with this PR.

File: mteb/tasks/Classification/pol/PolHateClassification.py (hunk 1 above)
Comment (dokato) [reply mteb/tasks/Classification/pol/PolHateClassification.py:72]: Regarding the last point, this is how the split looks like (alreday stratified by labels). Is that alright?
```
DatasetDict({
    train: Dataset({
        num_rows: 1907
    })
    test: Dataset({
        num_rows: 1908
    })
})
```

File: mteb/tasks/Classification/pol/PolHateClassification.py (hunk 2 above)
Comment (dokato) [reply mteb/tasks/Classification/pol/PolHateClassification.py:73]: I'm gonna leave it then casue as you say it's a good practice to clean up the unrelevant columns.

File: mteb/tasks/Classification/pol/PolHateClassification.py (hunk 3 above)
Comment (dokato) [reply mteb/tasks/Classification/pol/PolHateClassification.py]: removed

**Comment by dokato** [comment]
hey @imenelydiaker thank you for your insightful review. PTAL at the fixed version of it.
BTW I'm new to the project, so  I used `if __name__ == "__main__":` approach, because when I try to run the:

```
mteb -m intfloat/multilingual-e5-small -t PolHateClassification 
```

command it complains that:

```
WARNING:mteb.evaluation.MTEB:WARNING: Unknown tasks: PolHateClassification. Known tasks: AFQMC,AILACasedocs,AILAStatutes[...]
```
and finishes with `Evaluating 0 tasks:`. It still generates results okay with my approach, but maybe I'm missing something to make it fully running, so if you could point me? Thanks!

**Comment by imenelydiaker** [comment]
> hey @imenelydiaker thank you for your insightful review. PTAL at the fixed version of it. 

Thanks for your contribution!
>BTW I'm new to the project, so I used `if __name__ == "__main__":` approach, because when I try to run the:
> 
> ```
> mteb -m intfloat/multilingual-e5-small -t PolHateClassification 
> ```
> 
> command it complains that:
> 
> ```
> WARNING:mteb.evaluation.MTEB:WARNING: Unknown tasks: PolHateClassification. Known tasks: AFQMC,AILACasedocs,AILAStatutes[...]
> ```
> 
> and finishes with `Evaluating 0 tasks:`. It still generates results okay with my approach, but maybe I'm missing something to make it fully running, so if you could point me? Thanks!

Regarding this, if you'd like to use the CLI, you'll have to install the current package with your modifications using `make install` (it will run `pip install -e ".[dev]"`), and it will be able to find your newly created task.

**Comment by dokato** [comment]
thanks @imenelydiaker, this worked! got the same results, so no update was needed

**Comment by rbroc** [comment]
hi there, just noting that this will add a dataset that will already be included in #439, opened as a response to #395

**Comment by KennethEnevoldsen** [comment]
Ahh always a shame with duplicate work (we did try to prevent this by creating an issue for it, but understand that It will happen). Will close this issue, but @dokato we would love your submission on another dataset (you might want to check out: https://github.com/embeddings-benchmark/mteb/issues/419, but I encourage that you check if no other PR is open for the datasets)
